src/CabanaPD_HeatTransfer.hpp

```diff
@@ -0,0 +1,123 @@
+/****************************************************************************
+ * Copyright (c) 2022-2023 by Oak Ridge National Laboratory                 *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of CabanaPD. CabanaPD is distributed under a           *
+ * BSD 3-clause license. For the licensing terms see the LICENSE file in    *
+ * the top-level directory.                                                 *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#ifndef HEATTRANSFER_H
+#define HEATTRANSFER_H
+
+#include <CabanaPD_Timer.hpp>
+
+namespace CabanaPD
+{
+
+// Peridynamic heat transfer with forward-Euler time integration.
+// Inherits only because this is a similar neighbor-based kernel.
+template <class ExecutionSpace, class ModelType>
+class HeatTransfer : public Force<ExecutionSpace, BaseForceModel>
+{
+  protected:
+    using base_type = Force<ExecutionSpace, BaseForceModel>;
+    using base_type::_half_neigh;
+    using base_type::_timer;
+    Timer _euler_timer = base_type::_energy_timer;
+    ModelType _model;
+
+  public:
+    HeatTransfer( const bool half_neigh, const ModelType model )
+        : base_type( half_neigh )
+        , _model( model )
+    {
+    }
+
+    template <class TemperatureType, class PosType, class ParticleType,
+              class NeighListType, class ParallelType>
+    void
+    computeHeatTransferFull( TemperatureType& prev_temp, const PosType& x,
+                             const PosType& u, const ParticleType& particles,
+                             const NeighListType& neigh_list, const int n_local,
+                             ParallelType& neigh_op_tag )
+    {
+        _timer.start();
+
+        auto model = _model;
+        const auto vol = particles.sliceVolume();
+        const auto temp = particles.sliceTemperature();
+
+        auto temp_func = KOKKOS_LAMBDA( const int i, const int j )
+        {
+            double xi, r, s;
+            getDistance( x, u, i, j, xi, r, s );
+
+            const double coeff = model.thermal_coeff *
+                                 model.conductivity_function( xi ) / model.cp;
+            prev_temp( i ) +=
```

Comment:
Does this perform the summation over neighbors? If so, we may need a different naming?

Reply:
Not sure what is the best choice. However, this should correspond to the diffusive term in the heat equation (see: https://reference.wolfram.com/language/PDEModels/tutorial/HeatTransfer/HeatTransfer.html#1445086126).  I have seen named "nonlocal diffusion term" (e.g., file:///Users/psc/Downloads/mathematics-08-02111.pdf) (note, in our case, it is a nonlocal heat conduction equation, also referred to as "peridynamic formulation for heat conduction" or "peridynamic thermal diffusion" equation.)

Perhaps a variable name related to "nonlocal diffusion"? Something like "nonlocal_diff( i )"?

Reply:
I rename to `temperatureConduction` and locally just `conduction`